smoke/features/steps/steps.py

```diff
@@ -180,3 +176,38 @@ def pod_log_not_contains(context, pod_name, data, share_resource):
     if not match:
         print(f"pod {pod_name} do not reflect changes as refreshResource is disabled")
 
+@given(u'user has created resource type with shared resource')
+def data_driven_resource_create(context):
+    model = getattr(context, "model", None)
+    if not model:
+        context.model = ResourceTypeModel()
+    for row in context.table:
+        context.model.add_resource(row["resource type"], shareResource=row["shared resource"])
+        resources = context.model.get_resource_type(row["shared resource"])
+        if resources == "configmap":
+            configmap(context, "share-config")
+            shared_configmap(context, "my-shared-config", "share-config")
+        else:
+            create_secret(context, "my-secret")
+            shared_secret(context, "my-shared-secret", "my-secret")
+    
+@when(u'creates a pod {pod_name} that used the sharedconfigmaps or sharedsecrets')
+def create_pods(context, pod_name):
+    # Lets create a pod for configmap as an example
+    create_pod(context, pod_name, "sharedconfigmap", "true")
+
+@when(u'user removes the share related rbac "use" permissions')
+def delete_rbac(context):
+    namespace = Project().current_project()
+    name = "shared-resource-my-shared"
+    oc.delete("role", name, namespace)
+    oc.delete("rolebinding", name, namespace)
+
+@then(u'pod {pod_name} in the second project should remove the share related data from its volume')
+def remove_data_with_permission_removal(context, pod_name):
```

Comment:
python is by no means my strength, but I don't see any calls to this method

is the coming in an update to the PR?  or if it is already here, can you point me to it?

ultimately, I'm trying to see where you wait the sufficient amount of time for this delete to occur.... again, that delete is not instantaneous.  It occurs when the relist happens, where we re-execute the k8s subject access review requests for each pod consuming a share.

Reply:
> python is by no means my strength, but I don't see any calls to this method

ok, so these are called as `step functions` in behave and it gets identified by some decorator(@given, @when ...) defined by behave internally, then gets executed. More info: https://python-behave.readthedocs.io/en/latest/api.html#step-functions 
We want to be specific with name so that in future we can use this functionality or modify according. 

> ultimately, I'm trying to see where you wait the sufficient amount of time for this delete to occur.... again, that delete is not instantaneous. It occurs when the relist happens, where we re-execute the k8s subject access review requests for each pod consuming a share.

I have defined one function to check the [pod continuously with the interval of timeouts](https://github.com/openshift/csi-driver-shared-resource/pull/89/files#diff-763c955e01f2ae4d89f5656da619638e3786e7873921d84ec8b719ba3fb52381R243-R255) and that has been called in this step definition [here](https://github.com/openshift/csi-driver-shared-resource/pull/89/files#diff-bdd382db32c639a268fb0b941435828272be3d2389a7a2a4c822bbb768af28b3R211)
Once the data has been removed it will come out of loop execution. Here it executes for 10 attempts with 1 min of timeout. I hope it answers your question @gabemontero , if not please let me know.